src/main/resources/static/js/app.js

```diff
@@ -0,0 +1,194 @@
+    addCartItem(item){
+        const div = document.createElement('div');
+        div.classList.add('cart-item');
+        div.innerHTML = `<img src=${item.cover} alt="product">
+        <div>
+            <h4>${item.title}</h4>
+            <h5>${item.price}PLN</h5>
+            <span class="remove-item" data-id=${item.id}>usuń</span>
+        </div>
+        <div>
+            <i class="fas fa-chevron-up" data-id=${item.id}></i>
+            <p class="item-amount">${item.amount}</p>
+            <i class="fas fa-chevron-down" data-id=${item.id}></i>
+        </div>`;
+        cartContent.appendChild(div);
+        console.log(cartContent);
+    }
+    showCart(){
+        cartOverlay.classList.add('transparentBcg');
+        cartDOM.classList.add('showCart');
+    }
+    setupAPP(){
+        cart = Storage.getCart();
+        this.setCartValues(cart);
+        this.populateCart(cart);
+        cartBtn.addEventListener('click',this.showCart);
+        closeCartBtn.addEventListener('click',this.hideCart);
+    }
+    populateCart(cart){
+        cart.forEach(item => this.addCartItem(item));
+    }
+    hideCart(){
+        cartOverlay.classList.remove('transparentBcg');
+        cartDOM.classList.remove('showCart');
+    }
+    cartLogic(){
+        clearCartBtn.addEventListener("click",()=>{this.clearCart();});
+        cartContent.addEventListener('click', event=>{
+            if(event.target.classList.contains('remove-item')){
+                let removeItem = event.target;
+                let id = removeItem.dataset.id;
+                cartContent.removeChild(removeItem.parentElement.parentElement);
+                this.removeItem(id);
+            }else if(event.target.classList.contains('fa-chevron-up')){
+                let addAmount = event.target;
+                let id = addAmount.dataset.id;
+                let tempItem = cart.find(item => String(item.id) === id);
+                tempItem.amount = tempItem.amount + 1;
+                Storage.saveCart(cart);
+                this.setCartValues(cart);
+                addAmount.nextElementSibling.innerText = tempItem.amount;
+            }else if(event.target.classList.contains('fa-chevron-down')){
+                let lowerAmount = event.target;
+                let id  = lowerAmount.dataset.id;
+                let tempItem = cart.find(item => String(item.id) === id);
+                tempItem.amount = tempItem.amount - 1;
+                if (tempItem.amount > 0){
+                    Storage.saveCart(cart);
+                    this.setCartValues(cart);
+                    lowerAmount.previousElementSibling.innerText = tempItem.amount;
+                }else{
+                    cartContent.removeChild(lowerAmount.parentElement.parentElement);
+                    this.removeItem(id);
+                }
+
+            }
+        })
+    }
+    clearCart(){
+        let cartItems = cart.map(item => String(item.id));
+        cartItems.forEach(id => this.removeItem(id));
```

Comment:
cart.map(item => String(item.id)).forEach(id => this.removeItem(id))

Reply:
bez nowej zmiennej

Reply:
Zrobione